app/models/cookbook.rb

```diff
@@ -138,16 +139,21 @@ def sorted_cookbook_versions
   #
   def transfer_ownership(initiator, recipient)
     if initiator.is?(:admin) || collaborator_users.include?(recipient)
-      collaborator = Collaborator.new(user_id: user_id, resourceable: self)
       update_attribute(:user_id, recipient.id)
-      collaborator.save!
+
+      unless collaborators.where(user_id: initiator.id).present?
+        collaborator = Collaborator.new(user_id: initiator.id, resourceable: self)
+        collaborator.save!
+      end
 
       if collaborator_users.include?(recipient)
         collaborator = collaborators.where(
           user_id: recipient.id,
           resourceable: self
         ).first
-        collaborator.destroy unless collaborator.nil?
+        # Do not destroy collaborator is collaborator does not exist
+        # OR if the collaborator is associated with a group
+        collaborator.destroy unless collaborator.nil? || collaborator.group_id.present?
```

Comment:
Do not destroy a collaborator during an ownership transfer if the collaborator is associated with a group